mlir/python/requirements.txt

```diff
@@ -1,4 +1,4 @@
 numpy>=1.19.5, <=1.26
 pybind11>=2.9.0, <=2.10.3
-PyYAML>=5.3.1, <=6.0.1
-ml_dtypes   # provides several NumPy dtype extensions, including the bf16
\ No newline at end of file
+PyYAML>=5.4, <=6.0.1
+ml_dtypes   # provides several NumPy dtype extensions, including the bf16
```

Comment:
I know your change isn't adding this, but for the same security reason we want to update the PyYAML version, this should have a range or a specific version.

Reply:
@xurui1995 : We can add a reasonable version range here unless you let us know what you would prefer.

Reply:
ml_dtypes was added around end of May. I see that the latest mldtypes is 0.4.0 which was released on April 1st, so assume that we want at least 0.4.0 for now.

Reply:
At least 0.4.0 seems reasonable，we changed it to at most 0.4.0 now. Any reason? If it’s the requirement of pyyaml I can do a quick test for giving you a minimal version of ml_dtypes

Reply:
We need an upper bound to prevent potential attacks where new versions have vulnerabilities.

Reply:
Maybe we can go for `>=0.1.0 <=0.4.0 `, I tried 0.1.0 it still works for the test case in mlir.  For versions under 0.1.0 I think we can ignore them

Reply:
I have updated to `>=0.1.0, <=0.4.0` as suggested